docs/guide/vite.md

```diff
@@ -1,10 +1,10 @@
 # Vite
 
-[Vite](https://vite.dev/) is a modern build tool that significantly improves the development experience. Unlike traditional bundlers like webpack, Vite leverages native ES modules in the browser to enable lightning-fast hot module replacement (HMR) and on-demand compilation. During development, this means your changes appear instantly in the browser without rebuilding the entire bundle. When building for production, Vite uses Rollup to create highly optimized assets.
+[Vite](https://vite.dev/) is a modern build tool that significantly improves the development experience. Unlike traditional bundlers like webpack, Vite leverages native ES modules in the browser to enable faster builds and updates. During development, when using UI frameworks that support Hot Module Replacement (HMR) like Vue, React, or Svelte, your changes appear instantly in the browser without rebuilding the entire bundle or refreshing the page. For vanilla JavaScript or frameworks without HMR support, Vite still provides fast page reloads on changes. When building for production, Vite uses Rollup to create highly optimized assets.
 
 For ApostropheCMS projects, this translates to faster development cycles, improved debugging with better source maps, and a more streamlined build process that requires less configuration. The dev server starts up instantly regardless of your application size, and HMR updates happen in milliseconds rather than seconds.
 
-Using Vite in an ApostropheCMS project, rather than the default webpack build, is currently in beta. However, moving forward, webpack will slowly be deprecated. At this time we encourage you to test the new build in your `dev` environment.
+Using Vite in an ApostropheCMS project, rather than the default webpack build, is currently in beta. However, moving forward, webpack will slowly be deprecated. At this time we encourage you to test the new build in your `dev` environment. If you want to take advantage of HMR in your project-level non-Admin UI code, we recommend you start any new project using ESM modules in your frontend code. You can see the steps to begin this migration in our [documentation](/guide/migrating-to-esm). While HMR will work out of the box for the Admin UI with `hmr: 'apos'`, project-level UI code will need to use ESM syntax to get the full benefits of using Vite. Traditional CommonJS (require/module.exports) code will still get page reloads on changes, but won't support true HMR.
```

Comment:
Traditional CommonJS actually breaks Vite and Rollup - it will break builds :)

Reply:
Keep in mind we are talking only about "sources", the files that are considered non-bundled project level code. `npm` packages that are commonjs are handled/transformed properly by Vite. More information is avialble in the Vite docs (this is their limitation, not ours).

Reply:
Changed. 🤷🏻‍♂️